assets/src/edit-story/components/panels/elementAlignment.js

```diff
@@ -0,0 +1,366 @@
+          x,
+          y,
+          width,
+          height
+        );
+        offsetY = (frameHeight - height) / 2;
+      }
+      return {
+        y: boundRect.startY + offsetY,
+      };
+    });
+  };
+
+  const handleAlignMiddle = () => {
+    const centerY = (boundRect.endY + boundRect.startY) / 2;
+    onSetProperties((properties) => {
+      const { height } = properties;
+      return {
+        y: centerY - height / 2,
+      };
+    });
+  };
+
+  const handleAlignBottom = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offsetY = 0;
+      if (rotationAngle) {
+        const { height: frameHeight } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offsetY = (frameHeight - height) / 2;
+      }
+      return {
+        y: boundRect.endY - height - offsetY,
+      };
+    });
+  };
+
+  const handleHorizontalDistribution = () => {
+    updatedSelectedElementsWithFrame.sort(
+      (a, b) => (a.frameX + a.frameWidth) / 2 - (b.frameX + b.frameWidth) / 2
+    );
+    const commonSpaceWidth = dataPixels(
+      (boundRect.width -
+        updatedSelectedElementsWithFrame.reduce(
+          (sum, element) => sum + element.frameWidth,
+          0
+        )) /
+        (updatedSelectedElementsWithFrame.length - 1)
+    );
+    const updatedX = {};
+    let offsetX = 0;
+    updatedSelectedElementsWithFrame.forEach((element, index) => {
+      const { id, x, width, frameWidth } = element;
+      if (
+        index === 0 ||
+        index === updatedSelectedElementsWithFrame.length - 1
+      ) {
+        updatedX[id] = { x };
+        offsetX = x;
+      } else {
+        updatedX[id] = {
+          x: offsetX + (frameWidth - width) / 2,
+        };
+      }
+      offsetX += frameWidth + commonSpaceWidth;
+    });
+    onSetProperties(({ id }) => updatedX[id]);
+  };
+
+  const handleVerticalDistribution = () => {
+    updatedSelectedElementsWithFrame.sort(
+      (a, b) => (a.frameY + a.frameHeight) / 2 - (b.frameY + b.frameHeight) / 2
+    );
+    const commonSpaceHeight = dataPixels(
+      (boundRect.height -
+        updatedSelectedElementsWithFrame.reduce(
+          (sum, element) => sum + element.frameHeight,
+          0
+        )) /
+        (updatedSelectedElementsWithFrame.length - 1)
+    );
+    const updatedY = {};
+    let offsetY = 0;
+    updatedSelectedElementsWithFrame.forEach((element, index) => {
+      const { id, y, height, frameHeight } = element;
+      if (
+        index === 0 ||
+        index === updatedSelectedElementsWithFrame.length - 1
+      ) {
+        updatedY[id] = { y };
+        offsetY = y;
+      } else {
+        updatedY[id] = {
+          y: offsetY + (frameHeight - height) / 2,
+        };
+      }
+      offsetY += frameHeight + commonSpaceHeight;
+    });
+    onSetProperties(({ id }) => updatedY[id]);
+  };
```

Comment:
Basically they're doing same things. And we can use all the conditions on each steps, but I think it will make the code more difficult to read and understand. Top, left, right, bottom works fine and we can merge them. But I think these two maybe better to leave them as they like now.